src/main/java/org/spectrumauctions/sats/core/model/lsvm/LSVMGrid.java

```diff
@@ -90,24 +92,26 @@ LSVMLicense getLicense(int row, int column) {
     }
 
     List<LSVMLicense> getProximity(LSVMLicense center, int proximitySize) {
-        List<LSVMLicense> proximity = new LinkedList<>();
+    	// no duplicates 
+    	// use LinkedHashSet -> insertion order
+        HashSet<LSVMLicense> proximity = new LinkedHashSet<>();
         proximity.add(center);
-        List<LSVMLicense> tmp = expand(proximity, center);
+        HashSet<LSVMLicense> tmp = expand(proximity, center);
         // Expand step by step
         for (int i = 0; i < proximitySize; i++) {
             proximity.addAll(tmp);
-            List<LSVMLicense> tmp2 = new LinkedList<>();
+            HashSet<LSVMLicense> tmp2 = new LinkedHashSet<>();
             for (LSVMLicense license : tmp) {
                 tmp2.addAll(expand(proximity, license));
             }
             tmp = tmp2;
         }
-        return proximity;
+        return proximity.stream().collect(Collectors.toList());
```

Comment:
Are you 100% sure that this maintains the order in all cases? Just out of interest.

Reply:
Yes the Spliterator of the LinkedHashSet is ORDERED and so will be the stream:
https://docs.oracle.com/javase/10/docs/api/java/util/LinkedHashSet.html#spliterator()

Reply:
Great, thanks for the reference! :-D